worlds/yugioh06/__init__.py

```diff
@@ -0,0 +1,445 @@
+            "final_campaign_boss_challenges": self.options.final_campaign_boss_challenges.value,
+            "fourth_tier_5_campaign_boss_challenges": self.options.fourth_tier_5_campaign_boss_challenges.value,
+            "third_tier_5_campaign_boss_challenges": self.options.third_tier_5_campaign_boss_campaign_opponents.value,
+            "final_campaign_boss_campaign_opponents": self.options.final_campaign_boss_campaign_opponents.value,
+            "fourth_tier_5_campaign_boss_campaign_opponents": self.options.fourth_tier_5_campaign_boss_unlock_condition.value,
+            "third_tier_5_campaign_boss_campaign_opponents": self.options.third_tier_5_campaign_boss_campaign_opponents.value,
+            "number_of_challenges": self.options.number_of_challenges.value,
+        }
+
+        slot_data["removed challenges"] = self.removed_challenges
+        slot_data["starting_booster"] = self.starting_booster
+        slot_data["starting_opponent"] = self.starting_opponent
+        return slot_data
+
+    def interpret_slot_data(self, slot_data: Dict[str, Any]) -> None:
+        # bypassing random yaml settings
+        self.options.structure_deck.value = slot_data["structure_deck"]
+        self.options.banlist.value = slot_data["banlist"]
+        self.options.final_campaign_boss_unlock_condition.value = slot_data["final_campaign_boss_unlock_condition"]
+        self.options.fourth_tier_5_campaign_boss_unlock_condition.value = \
+            slot_data["fourth_tier_5_campaign_boss_unlock_condition"]
+        self.options.third_tier_5_campaign_boss_unlock_condition.value = \
+            slot_data["third_tier_5_campaign_boss_unlock_condition"]
+        self.options.final_campaign_boss_challenges.value = \
+            slot_data["final_campaign_boss_challenges"]
+        self.options.fourth_tier_5_campaign_boss_challenges.value = \
+            slot_data["fourth_tier_5_campaign_boss_challenges"]
+        self.options.third_tier_5_campaign_boss_challenges.value = \
+            slot_data["third_tier_5_campaign_boss_challenges"]
+        self.options.final_campaign_boss_campaign_opponents.value = \
+            slot_data["final_campaign_boss_campaign_opponents"]
+        self.options.fourth_tier_5_campaign_boss_campaign_opponents.value = \
+            slot_data["fourth_tier_5_campaign_boss_campaign_opponents"]
+        self.options.third_tier_5_campaign_boss_campaign_opponents.value = \
+            slot_data["third_tier_5_campaign_boss_campaign_opponents"]
+        self.options.number_of_challenges.value = \
+            slot_data["number_of_challenges"]
+        self.removed_challenges = slot_data["removed challenges"]
+        self.starting_booster = slot_data["starting_booster"]
+        self.starting_opponent = slot_data["starting_opponent"]
+        all_state = self.multiworld.get_all_state(False)
+
+        return all_state
```

Comment:
If you return something with interpret_slot_data, it regens with the information you return at multiworld.re_gen_passthrough["{game_name}"]. Currently, I don't see anywhere you're doing that, so a return is unnecessary. And also, returning all state is definitely incorrect here.

Would recommend asking in the universal tracker thread for some more info on getting this working, or looking at how Tunic implements the same idea of bypassing the yaml settings

Reply:
This used to work, guess i haven't checked in a while.
Anyways it should work now.